### ISSUE TITLE
Ensure privacy of users is respected with exception notifications

### ISSUE THREAD

**Issue by NSkelsey** [globaleaks/GlobaLeaks#1799]
In order to quickly diagnose potential problems in the software when exceptions in clients are generated, they are automatically reported to the backend. The backend server caches these exceptions and sends them to the server administrator via email. 

In order to prevent inadvertent information leaks the logs should be run through filters that redact the potential inclusion of: (x marks implemented):

- sessions keys
- email addresss (x)
- pgp formatted strings
- uuids (x)

**Comment by sunilpateliit** [comment]
Hi @NSkelsey ! Is this issue still open ? If yes then I would like to work on it. Thanks.

**Comment by evilaliv3** [comment]
Dear @sunilpateliit , thank you for your question.

Yes this is still a valid ticket where to work on and still open.

Do you have already ideas that lead you to select this ticket or you are just looking for a ```good first issue``` in order to contribute to GlobaLeaks? I'm asking this as this ticket is actually open to keep track of possible improvement without being nor so immediate in the real.

Thank you again for your interest in contributing to the project; I invite you to join our slack at https://slack.globaleaks.org where you find us and our community.

**Comment by sunilpateliit** [comment]
Hi @evilaliv3, 
Thanks for the quick response. I picked this issue in order to contribute to GlobaLeaks also I've the following idea to resolve this issue: 
filter the **value** for any **sessions keys** and **pgp formatted strings** before calling mail_exception_handler in [state.py](https://github.com/globaleaks/GlobaLeaks/blob/main/backend/globaleaks/state.py#L297). 

Please let me know if there are any drawbacks in this approach.

**Comment by evilaliv3** [comment]
Thank you the approach is fine and would definitely be an improvement!

We may consider moving there inside state.py as well the code that currently was loaded into the client https://github.com/globaleaks/GlobaLeaks/blob/main/client/app/js/app.js#L51

In fact the backend component is a trusted element in the threat model and would be able anyhow to temper this filter so i francly think a unified approach on the backend would be better.

what do you think?

**Comment by sunilpateliit** [comment]
I agree with you on the unified approach on the backend to be a better solution. Shall I make the changes in the backend code and raise the pr ?

**Comment by evilaliv3** [comment]
Yes please do always feel free to propose your contributions!

As for you patch please consider to make a pull request against the devel
branch
